File: typescript/word-count/word-count.ts
export default class Words {
  count(text: string): Map<string, number> {
    return text
      .trim()
      .toLowerCase()
      .split(/[ \n\t]+/)
      .reduce((map, word) => map.set(word, (map.get(word) || 0) + 1), new Map<string, number>());
  }
}

File: go/word-count/word_count.go
package wordcount

import (
	"regexp"
	"strings"
)

var wordRegex = regexp.MustCompile(`\b(\d+|[a-z]|[a-z][a-z']*[a-z])\b`)

// Frequency stores how many times words appear in a phrase.
type Frequency map[string]int

// WordCount counts the number of occurrences of each word in the given phrase.
func WordCount(phrase string) Frequency {
	f := Frequency{}
	for _, w := range wordRegex.FindAllString(strings.ToLower(phrase), -1) {
		f[w]++
	}
	return f
}
